src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -66,6 +66,7 @@ class Container implements ResettableContainerInterface
     protected $services = array();
     protected $methodMap = array();
     protected $privates = array();
+    protected $synthetics = array();
```

Comment:
Method for synthetic services throw exceptions, which means they are functionally useless. So for me it's safe to remove them from methodMap yes.